src/components/ha-control-circular-slider.ts

```diff
@@ -0,0 +1,481 @@
+        if (this._activeSlider) {
+          fireEvent(this, `${this._activeSlider}-changing`, {
+            value: undefined,
+          });
+          fireEvent(this, `${this._activeSlider}-changed`, { value: stepped });
+        }
+        this._activeSlider = undefined;
+      });
+    }
+  }
+
+  private get _tenPercentStep() {
+    return Math.max(this.step, (this.max - this.min) / 10);
+  }
+
+  private _handleKeyDown(e: KeyboardEvent) {
+    if (!A11Y_KEY_CODES.has(e.code)) return;
+    e.preventDefault();
+    this._activeSlider = (e.currentTarget as any).id as ActiveSlider;
+
+    const value = this._getActiveValue();
+
+    switch (e.code) {
+      case "ArrowRight":
+      case "ArrowUp":
+        this._setActiveValue(
+          this._boundedValue((value ?? this.min) + this.step)
+        );
+        break;
+      case "ArrowLeft":
+      case "ArrowDown":
+        this._setActiveValue(
+          this._boundedValue((value ?? this.min) - this.step)
+        );
+        break;
+      case "PageUp":
+        this._setActiveValue(
+          this._steppedValue(
+            this._boundedValue((value ?? this.min) + this._tenPercentStep)
+          )
+        );
+        break;
+      case "PageDown":
+        this._setActiveValue(
+          this._steppedValue(
+            this._boundedValue((value ?? this.min) - this._tenPercentStep)
+          )
+        );
+        break;
+      case "Home":
+        this._setActiveValue(this._boundedValue(this.min));
+        break;
+      case "End":
+        this._setActiveValue(this._boundedValue(this.max));
+        break;
+    }
+    fireEvent(this, `${this._activeSlider}-changing`, { value: this.value });
+    this._activeSlider = undefined;
+  }
+
+  _handleKeyUp(e: KeyboardEvent) {
+    if (!A11Y_KEY_CODES.has(e.code)) return;
+    this._activeSlider = (e.currentTarget as any).id as ActiveSlider;
+    e.preventDefault();
+    fireEvent(this, "value-changed", { value: this.value });
+    this._activeSlider = undefined;
+  }
+
+  destroyListeners() {
+    if (this._mc) {
+      this._mc.destroy();
+      this._mc = undefined;
+    }
+  }
+
+  protected render(): TemplateResult {
+    const currentAngle = 160;
+
+    const trackPath = arc({ x: 0, y: 0, start: 0, end: MAX_ANGLE, r: 150 });
+
+    const maxRatio = MAX_ANGLE / 360;
+
+    const f = 150 * 2 * Math.PI;
+    const lowValue = this.dual ? this.low : this.value;
+    const highValue = this.high;
+    const lowPercentage = this._valueToPercentage(lowValue ?? this.min);
+    const highPercentage = this._valueToPercentage(highValue ?? this.max);
+
+    const lowArcLength = lowPercentage * f * maxRatio;
+    const lowStrokeDasharray = `${lowArcLength} ${f - lowArcLength}`;
+
+    const highArcLength = (1 - highPercentage) * f * maxRatio;
+    const highStrokeDasharray = `${highArcLength} ${f - highArcLength}`;
+    const highStrokeDashOffset = `${highArcLength + f * (1 - maxRatio)}`;
+
+    return svg`
```

Comment:
```suggestion
    return html`
```

`svg` is for when used inside a `svg` element